modules/tensor_mechanics/doc/content/syntax/Modules/TensorMechanics/CohesiveZoneMaster/index.md

```diff
@@ -1,35 +1,90 @@
-!syntax parameters /Modules/TensorMechanics/CohesiveZoneMaster/CohesiveZoneMasterAction
-!syntax inputs /Modules/TensorMechanics/CohesiveZoneMaster/CohesiveZoneMasterAction
+1. [CZMComputeDisplacementJumpTotalLagrangian](CZMComputeDisplacementJumpTotalLagrangian.md)
+2. [CZMComputeGlobalTractionTotalLagrangian](CZMComputeGlobalTractionTotalLagrangian.md)
+3. [CZMInterfaceKernelTotalLagrangian](CZMInterfaceKernelTotalLagrangian.md)
 
-### Large Deformations
+As mentioned in previous section, the `CohesiveZoneMaster` action adds the appropriate objects depending on the selected kinematic formulation.  
 
-Large deformations have not been implemented yet. An error occurs when the user tries to manually set `use_displaced_mesh = true` in a  `CZMMaterial` or in a `CZMInterfaceKernel`.
+!alert warning
+To obtain a traction consistent with the bulk stress, the `Small Strain` kinematic should be used together with a bulk `Small Strain` formulation, and the `Total Lagrangian` kinematics should be used together with a `Finite Strain` formulation.
+
+### Cohesive Zone Constitutive Models
+
+The implemented framework allows to implement two different types of cohesive zone constitutive models (e.g. traction separation laws): i) path independent, and ii) path dependent.
+The first type of models assumes the traction is only function of the total interface  displacement jump, i.e. $\hat{t} =f\left(\llbracket \hat{u} \rrbracket\right)$. This kind of models do not have history variables and can be implemented overriding the `ComputeLocalTractionTotalBase` class.
+Instead, path dependent models have history variables and assume the traction increment is a function of the interface displacement jump increment, and its old value, $\Delta\hat{t} =f\left(\Delta \llbracket \hat{u} \rrbracket, \llbracket \hat{u} \rrbracket_{old} \right)$. Path dependent models can be implemented overriding the `ComputeLocalTractionIncrementalBase` class.
+In both cases, the traction separation law should always be formulated in terms of the local interface response and assuming `Small strain`. Furthermore, all constitutive laws should be implemented for the 3D general case, even if used for 1D or 2D simulations.
+The kinematic is already embedded in the `ComputeDisplacementJump` and `ComputeGlobalTraction` objects.
+Both types of `ComputeLocalTraction` objects allow using either the `Small Strain` and the `Total Lagrangian` kinematics.
+
+## Input Examples
+
+The following example show how to use the `CohesiveZoneMaster` action.
+
+!listing modules/tensor_mechanics/test/tests/cohesive_zone_model/stretch_rotate_large_deformation.i block=Modules/TensorMechanics/CohesiveZoneMaster
+
+If necessary, multiple instances of the `CohesiveZoneMaster` action can be added, for instance when different material properties `base_name` are needed for different boundaries. The `base_name` parameter used in the action should also be provided to the associated materials.
+
```

Comment:
Could you add  description of`generate_output` parameter here?

Reply:
done